application/src/main/java/org/togetherjava/tjbot/commands/moderation/ReportCommand.java

```diff
@@ -158,22 +157,38 @@ private MessageCreateAction createModMessage(String modalMessage, long userID,
                     [Reported Message URL](%s)
                     Modal Message: **%s**""").formatted(reportedMessage, reportedMessageUrl,
                     modalMessage))
-            .setAuthor("Author ID: %s".formatted(userID))
+            .setAuthor("Author ID: %d".formatted(userID))
             .setColor(AMBIENT_COLOR)
             .setTimestamp(Instant.now())
             .build();
 
         return modMailAuditLog.sendMessageEmbeds(embed);
     }
 
-    private void sendModMessage(InteractionHook hook, MessageCreateAction message) {
-        message.mapToResult().map(result -> {
-            if (result.isSuccess()) {
-                return "Thank you for reporting this message. A moderator will take care of the matter as soon as possible.";
-            }
-            logger.warn("Unable to forward a message report to modmail channel.");
-            return "Sorry, there was an issue sending your report to the moderators. We are investigating.";
-        }).flatMap(hook::editOriginal).queue();
+    private void sendModMessage(ModalInteractionEvent event, List<String> args,
+            TextChannel modMailAuditLog) {
+        InteractionHook hook = event.getHook();
+
+        String reportedMessage = args.get(0);
+        String reportedMessageUrl = args.get(1);
+
+        event.deferReply().setEphemeral(true).queue();
+        String modalMessage = event.getValue(REPORT_REASON).getAsString();
+        long userID = event.getUser().getIdLong();
+        createModMessage(modalMessage, userID, reportedMessage, reportedMessageUrl, modMailAuditLog)
+            .mapToResult()
+            .map(this::createUserReply)
+            .flatMap(hook::editOriginal)
+            .queue();
```

Comment:
See [this conversation](https://github.com/Together-Java/TJ-Bot/pull/653#discussion_r1007064480)